packages/core-flows/src/definition/cart/steps/update-carts.ts

```diff
@@ -0,0 +1,61 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  CartDTO,
+  FilterableCartProps,
+  ICartModuleService,
+  UpdateCartDataDTO,
+} from "@medusajs/types"
+import { getSelectsAndRelationsFromObjectArray } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+type UpdateCartsStepInput = {
+  selector: FilterableCartProps
+  update: UpdateCartDataDTO
+}
+
+export const updateCartsStepId = "update-carts"
+export const updateCartsStep = createStep(
+  updateCartsStepId,
+  async (data: UpdateCartsStepInput, { container }) => {
+    const service = container.resolve<ICartModuleService>(
+      ModuleRegistrationName.CART
+    )
+
+    const { selects, relations } = getSelectsAndRelationsFromObjectArray([
+      data.update,
+    ])
+
+    const prevCarts = await service.list(data.selector, {
+      select: selects,
+      relations,
+    })
+
+    const updatedCarts = await service.update(
+      data.selector as Partial<CartDTO>,
+      data.update
+    )
+
+    return new StepResponse(updatedCarts, prevCarts)
+  },
+  async (previousCarts, { container }) => {
+    if (!previousCarts?.length) {
+      return
+    }
+
+    const service = container.resolve<ICartModuleService>(
+      ModuleRegistrationName.CART
+    )
+
+    const toRestore = previousCarts.map((c) => ({
```

Comment:
todo: I think that we are missing the relations here, so it would partially work right?

Reply:
I think we are yet to figure out a good approach for restoring the correct relations no? Or have we settled on a pattern? Because it would depend on what you pass right

Reply:
we haven't discussed that at all so far. but we will need to think about it indeed

Reply:
Created a ticket for it